lib/p2p/Peer.ts

```diff
@@ -932,6 +934,12 @@ class Peer extends EventEmitter {
   }
 
   private handleDisconnecting = (packet: packets.DisconnectingPacket): void  => {
+    if (packet.body && packet.body.reason === DisconnectionReason.Banned) {
```

Comment:
I think this if statement should be moved within the if statement below. The main problem as I see it is that if we set `this.recvDisconnectionReason` here then we won't enter the next `if` statement and instead go to the `else` where we should penalize the peer. i'm thinking it should look like this:

```javascript
if (!this.recvDisconnectionReason && packet.body && packet.body.reason) {
  this.logger.debug(`received disconnecting packet from ${this.label}:${JSON.stringify(packet.body)}`);
  this.recvDisconnectionReason = packet.body.reason;
  if (packet.body.reason === DisconnectionReason.Banned) {
    // if we are banned by the peer set `bannedBy` in DB.
	const nodePubKey = this._nodePubKey || this.expectedNodePubKey;
	if (nodePubKey) {
	  this.emit('bannedBy', this.expectedNodePubKey!);
	}
  }
} else {
  // protocol violation: packet should be sent once only, with body, with `reason` field
  // TODO: penalize peer
}
```